xarray/backends/apiv2.py

```diff
@@ -202,12 +219,27 @@ def open_dataset(
     if engine is None:
         engine = _autodetect_engine(filename_or_obj)
 
+    decoders = resolve_decoders_kwargs(
+        decode_cf,
+        engine=engine,
+        mask_and_scale=mask_and_scale,
+        decode_times=decode_times,
+        decode_timedelta=decode_timedelta,
+        concat_characters=concat_characters,
+        use_cftime=use_cftime,
+        decode_coords=decode_coords,
+    )
+
     backend_kwargs = backend_kwargs.copy()
     overwrite_encoded_chunks = backend_kwargs.pop("overwrite_encoded_chunks", None)
 
-    open_backend_dataset = _get_backend_cls(engine, engines=ENGINES)
+    open_backend_dataset = _get_backend_cls(engine, engines=plugins.ENGINES)[
+        "open_dataset"
+    ]
     backend_ds = open_backend_dataset(
         filename_or_obj,
+        drop_variables=drop_variables,
+        **decoders,
         **backend_kwargs,
         **{k: v for k, v in kwargs.items() if v is not None},
     )
```

Comment:
For completeness, let me mention an alternative design that would not need `inspect.signature`. Instead of using a single function `open_backend_dataset`, we could use a "Loader" class with two separate methods, one for decoding a dataset and another for returning a raw dataset, e.g.,

```python
backend_cls = _get_backend_cls(engine)
loader = backend_loader(filename_or_obj, **backend_kwargs)
if decode:
    ds = loader.load_decoded(**decoders)
else:
    ds = loader.load_raw()
```

Is this better than the current approach in this PR (using inspect), or the current approach (on master) of calling the single `open_backend_dataset()`  function with `decode_cf=False`? Honestly I'm not sure. It is most explicit, but also probably a little more annoying to write.

Reply:
It seems to me that in this way we are going to complicate the interface without a real advantage. But I'm not sure about it. @alexamici, @jhamman what do you think about it?

Reply:
@shoyer the main reason we proposed the use `inspect` is to raise an appropriate error message when a backend  doesn't support a specific decoding option.

Your proposal simplifies the mangling of the decode options, but I'd still use `inspect` in the same as we do now before calling `load_decode`.

Personally I'd favour keeping the current implementation.